### ISSUE TITLE
Switching default profile breaks deployment

### ISSUE THREAD

**Issue by arundhaj** [aws/chalice#489]
In multiple AWS account scenario, when we change the default profile environment variable, it breaks the deployment. It is because, `deployed.json` doesn't have profile attributes.

The existing structure of `deployed.json` can be changed to contain profile name attribute, like below:
```
{
  "default": {
    "dev": {
      "api_handler_name": "",
      "api_handler_arn": "",
      ...
    },
  "profile1": {
    "dev": {
      "api_handler_name": "",
      "api_handler_arn": "",
      ...
    }
  }
}
```

**Comment by kyleknap** [comment]
I can confirm. I get the following error when I try to redeploy under different credentials:
```
$ chalice deploy
Regen deployment package.
Updating IAM policy for role: myapp-dev
Updating lambda function: myapp-dev
Initiating first time deployment.
ERROR - While deploying your chalice application, received the following error:

 An error occurred (BadRequestException) when calling the ImportRestApi 
 operation: Errors found during import:
        Unable to put integration on 
 'GET' for resource at path '/': Lambda function ARN must be in same account

```
Just to make sure is the expectation that Chalice will deploy a copy of the application under the credentials you provided?

I think it may be a little tricky to get this correct. While marking the profile does help in the `deployed.json` would help, it will not handle all of the edge cases. For example, if the user sets the credentials through environment variables (i.e. `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`), they would run into the same issue as those env variables are not associated to a profile.

**Comment by arundhaj** [comment]
You are right @kyleknap. Maybe, instead of using profile name, can we use hash of the credentials. Like;
```
{
  "<hash_of_credentials1>": {
    "dev": {
      "api_handler_name": "",
      "api_handler_arn": "",
      ...
    },
  "<hash_of_credentials2>": {
    "dev": {
      "api_handler_name": "",
      "api_handler_arn": "",
      ...
    }
  }
}
```

**Comment by dmulter** [comment]
I'm not quite using it yet either, but I do think Chalice needs to support usage for those that partition environments (dev, beta, prod, etc) across multiple AWS accounts. I"m debating the switch right now, but plan to delay until full Chalice support is provided.

**Comment by dperconti** [comment]
Curious if this ticket had been given any more thought?
